src/Common/src/Common.Expression/TypedValue.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Steeltoe.Common.Util;
+using System;
+
+namespace Steeltoe.Common.Expression
+{
+    public class TypedValue : ITypedValue
+    {
+        public static readonly TypedValue NULL = new TypedValue(null);
+
+        public TypedValue(object value)
+        {
+            Value = value;
+            TypeDescriptor = value?.GetType();
+        }
+
+        public TypedValue(object value, Type typeDescriptor)
+        {
+            Value = value;
+            TypeDescriptor = typeDescriptor;
+        }
+
+        public object Value { get; }
+
+        public Type TypeDescriptor { get; }
+
+        public override bool Equals(object other)
+        {
+            if (this == other)
+            {
+                return true;
+            }
+
+            if (!(other is TypedValue))
```

Comment:
```suggestion
            if (other is not TypedValue)
```

Reply:
This is for cosmetics/readability, it is not the only case and is completely optional